exp/api/v1alpha3/azuremanagedcontrolplane_webhook.go

```diff
@@ -90,11 +93,34 @@ func (r *AzureManagedControlPlane) ValidateDelete() error {
 	return nil
 }
 
+// Validate the Azure Managed ControlPlane and return an aggregate error
+func (r *AzureManagedControlPlane) Validate() error {
+	var allErrs field.ErrorList
+	err := r.validateDNSServiceIP()
+	if err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	if r.Spec.Version != "" {
```

Comment:
I don't think we need `if r.Spec.Version != ""`, an empty version is not a valid version